pages/cartPage.js

```diff
@@ -0,0 +1,296 @@
+            </Link>
+          ) : (
+            ' '
+          )}
+        </p>
+      </div>
+      <Footer />
+      <style jsx>{`
+        .container {
+          margin-top: 140px;
+          width: 80%;
+          margin-left: auto;
+          margin-right: auto;
+          background-color: white;
+          border-radius: 20px;
+          padding: 10px;
+        }
+        h1 {
+          margin-left: 15px;
+        }
+        a {
+          text-decoration: none;
+          font-size: 17px;
+          color: #2f3640;
+          margin-left: 20px;
+          font-weight: bold;
+        }
+
+        a:hover {
+          text-decoration: underline;
+        }
+        .backToShop {
+          background-color: #2ed573;
+          padding: 7px;
+          border-radius: 7px;
+        }
+        .backToShop:hover {
+          background-color: #2f3640;
+          color: white;
+          transition: background-color 0.3s;
+        }
+        .tableItems {
+          padding: 10px;
+
+          text-align: center;
+          border-bottom: 1px solid #2f3640;
+        }
+
+        .headings {
+          display: grid;
+          grid-template-columns: 1fr 2fr 1fr 1fr 1fr;
+          grid-gap: 10px;
+        }
+
+        .item {
+          display: grid;
+          grid-template-columns: 1fr 2fr 1fr 1fr 1fr;
+          grid-gap: 10px;
+        }
+
+        .buttonz {
+          display: flex;
+          flex-direction: row;
+          padding: 0px;
+          justify-content: center;
+          align-items: baseline;
+        }
+
+        img {
+          height: 70px;
+          align-self: center;
+          margin-left: 30px;
+        }
+        input {
+          padding: 5px;
+          border-radius: 5px;
+          width: 50px;
+          text-align: center;
+        }
+
+        .total {
+          text-align: right;
+          font-size: 20px;
+          margin-right: 10px;
+        }
+
+        button {
+          margin-top: 40px;
+          align-self: center;
+          background-color: #2ed573;
+          padding: 15px;
+          border-radius: 10px;
+          color: white;
+          font-family: inherit;
+          font-size: 120%;
+        }
+        button:hover {
+          background-color: #636e72;
+          transition: background-color 0.3s;
+        }
+      `}</style>
+      <style jsx global>{`
+        html,
+        body {
+          padding: 0;
+          margin: 0;
+          font-family: 'DM Mono', monospace;
+          background-image: url('/more-leaves.png');
+        }
+
+        * {
+          box-sizing: border-box;
+        }
+
+        h1 {
+          color: #2f3640;
+          text-shadow: 2px 2px black;
+        }
+      `}</style>
+    </div>
+  );
+}
+export default CartPage;
+
+export async function getServerSideProps(context) {
+  const { cart } = await nextCookies(context);
+  const { getProducts } = await import('../db.js');
+  const products = await getProducts(context.params);
```

Comment:
I guess (and everywhere else that you call this function) here you don't need to pass an argument.

Reply:
Additionally, maybe it makes sense to do this differently - pass all `id`s in your cart to a new function called `getProductsByIds`.